core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -383,11 +412,30 @@ class LogManager(val logDirs: Array[File],
         cleaner.abortCleaning(topicAndPartition)
         cleaner.updateCheckpoints(removedLog.dir.getParentFile)
       }
-      removedLog.delete()
-      info("Deleted log for partition [%s,%d] in %s."
-           .format(topicAndPartition.topic,
-                   topicAndPartition.partition,
-                   removedLog.dir.getAbsolutePath))
+      // renaming the directory to topic-partition.unique-Id.delete
+      val dirName = StringBuilder.newBuilder
+      dirName.append(removedLog.name)
+      dirName.append(".")
+      val uniqueId = java.util.UUID.randomUUID.toString.replaceAll("-","")
+      dirName.append(uniqueId)
+      dirName.append(Log.DeleteDirSuffix)
+      removedLog.close()
+      val renamed = new File(removedLog.dir.getParent, dirName.toString())
+      val renameSuccessful = removedLog.dir.renameTo(renamed)
+      if (renameSuccessful) {
+        removedLog.dir = renamed
+        // change the file pointers for log and index file
+        for (logSegment <- removedLog.logSegments) {
+          logSegment.log.file = new File(renamed, logSegment.log.file.getName)
+          logSegment.index.file = new File(renamed, logSegment.index.file.getName)
+        }
+
+        logsToBeDeleted synchronized {
+          logsToBeDeleted += removedLog
+        }
```

Comment:
Log a statement that log has been scheduled for deletion and name of the temporary directory.

Likewise [over here](https://github.com/apache/kafka/pull/636/files#diff-ffa8861e850121997a534ebdde2929c6R694) - can you add a log statement: `“Deleted %s in %d milliseconds"`